presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueExpressionUtil.java

```diff
@@ -13,72 +13,172 @@
+import com.facebook.presto.common.predicate.Range;
+import com.facebook.presto.common.predicate.ValueSet;
+import com.facebook.presto.common.type.DateType;
+import com.facebook.presto.common.type.TimestampType;
+import com.facebook.presto.common.type.Type;
 import com.facebook.presto.hive.metastore.Column;
-import com.facebook.presto.spi.PrestoException;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Joiner;
-import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
 
-import java.util.LinkedList;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 
-import static com.facebook.presto.hive.HiveErrorCode.HIVE_METASTORE_ERROR;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.HIVE_DEFAULT_DYNAMIC_PARTITION;
+import static com.facebook.presto.hive.metastore.MetastoreUtil.convertRawValueToString;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterators.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
 
 public final class GlueExpressionUtil
 {
-    private static final Joiner JOINER = Joiner.on(" AND ");
+    static final int GLUE_EXPRESSION_CHAR_LIMIT = 2048;
     private static final Set<String> QUOTED_TYPES = ImmutableSet.of("string", "char", "varchar", "date", "timestamp", "binary", "varbinary");
+    private static final String CONJUNCT_SEPARATOR = " AND ";
+    private static final Joiner CONJUNCT_JOINER = Joiner.on(CONJUNCT_SEPARATOR);
+    private static final String DISJUNCT_SEPARATOR = " OR ";
+    private static final Joiner DISJUNCT_JOINER = Joiner.on(DISJUNCT_SEPARATOR);
+
+    /**
+     * AWS Glue uses internally <a href="http://jsqlparser.sourceforge.net/home.php">JSQLParser</a>
+     * as SQL statement parser for the expression that can be used to filter the partitions.
+     * JSQLParser defines a set of reserved keywords which cannot be used as column names in the filter.
+     *
+     * @see <a href="https://sourceforge.net/p/jsqlparser/code/HEAD/tree/trunk/src/main/javacc/JSqlParserCC.jj">JSqlParser Grammar</a>
+     */
+    private static final Set<String> JSQL_PARSER_RESERVED_KEYWORDS = ImmutableSet.of(
+            "AS", "BY", "DO", "IS", "IN", "OR", "ON", "ALL", "AND", "ANY", "KEY", "NOT", "SET", "ASC", "TOP", "END", "DESC", "INTO", "NULL", "LIKE", "DROP", "JOIN",
+            "LEFT", "FROM", "OPEN", "CASE", "WHEN", "THEN", "ELSE", "SOME", "FULL", "WITH", "TABLE", "WHERE", "USING", "UNION", "GROUP", "BEGIN", "INDEX", "INNER",
+            "LIMIT", "OUTER", "ORDER", "RIGHT", "DELETE", "CREATE", "SELECT", "OFFSET", "EXISTS", "HAVING", "INSERT", "UPDATE", "VALUES", "ESCAPE", "PRIMARY",
+            "NATURAL", "REPLACE", "BETWEEN", "TRUNCATE", "DISTINCT", "INTERSECT");
 
     private GlueExpressionUtil() {}
 
+    private static boolean isQuotedType(Type type)
```

Comment:
Can you please reorder the methods in this file so that the order is: public, package private, private? Within each level, the order shall be the order they were called.